integration_test.go

```diff
@@ -374,11 +374,33 @@ func TestStop(t *testing.T) {
 	}
 }
 
-func writeDataIntoTestapp(t *testing.T, data string) {
-	err := ioutil.WriteFile("testapp/data/file", []byte(data), os.FileMode(0644))
+func atomicWriteFile(t *testing.T, data string, filename string) error {
+  tempFile, err := ioutil.TempFile(os.TempDir(), "tempData")
+	if err != nil {
+		return fmt.Errorf("could not create temp file: %s\n", err)
+	}
+
+	defer os.Remove(tempFile.Name())
+	defer tempFile.Close()
+
+	_, err = tempFile.Write([]byte(data))
+	if err != nil {
+		fmt.Errorf("Could not write to temp file: %s\n", err)
+	}
+
+	err = os.Rename(tempFile.Name(), filename)
 	if err != nil {
-		t.Fatalf("write file: %s\n", err)
+		fmt.Errorf("Could not move temp file to testapp: %s\n", err)
 	}
+
+  return nil
+}
+
+func writeDataIntoTestapp(t *testing.T, data string) {
+	err := atomicWriteFile(t, data, "testapp/data/file")
```

Comment:
@barryskal you should run everything through go fmt (I've got a vim plugin that does this on save automatically)